api/v1alpha1/validation/envoygateway_validate.go

```diff
@@ -79,6 +83,31 @@ func validateEnvoyGatewayKubernetesProvider(provider *egv1a1.EnvoyGatewayKuberne
 	return nil
 }
 
+func validateEnvoyGatewayFileProvider(provider *egv1a1.EnvoyGatewayCustomProvider) error {
+	if provider == nil {
+		return fmt.Errorf("empty custom provider settings for file provider")
+	}
+
+	rType, iType := provider.Resource.Type, provider.Infrastructure.Type
+	if rType != egv1a1.ResourceProviderTypeFile || iType != egv1a1.InfrastructureProviderTypeHost {
```

Comment:
prefer if we didnt provide this check,  we may have a case where a sidecar hosts a web server and downloads resources and writes them into the directory, and the infra could continue to be provisioned in k8s

Reply:
sounds reasonable, so is this case a `file provider + k8s infra` combination ?